### PR TITLE
Shopping Cart Presistence

### PR DESCRIPTION
**Pull request by Tiewhan**
<h1>Overview</h1>
Added persistence logic to the shopping cart allowing the user to add items to the cart that will stay when the user closes the application

<h2>Additions Made</h2>
Added a repository for the shopping cart.
Added observers and observables to the view-model and model for the shopping cart.

### CODE REVIEW

File: Source/CoreDataFiles/CoreDataAppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
```
Comment (Tiewhan) [inline Source/CoreDataFiles/CoreDataAppDelegate.swift:1]: This is essentially a wrapper that the app delegate in the main project must conform to. It just allows the dev pod to make use of the app delegate without needing to depend on the main project

**Review by XanderSchoeman — CHANGES_REQUESTED** [review]
Just check that bitrise status checks* can pass. Then ill approve.

**Review by JancoDVTDev — APPROVED** [review]
Code looks clean and easy to read.

File: Source/ShoppingCart/Repositories/ShoppingCartCoreDataRepo.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    do {
+      
+      let list = try context.fetch(request)
+      
+      let shoppingCart = ShoppingCart()
+      shoppingCart.shoppingList = list.map { $0.mapToGame() }
+      
+      observer?.loadedData(of: shoppingCart)
+      
+    } catch { }
```
Comment (Zamo22) [inline Source/ShoppingCart/Repositories/ShoppingCartCoreDataRepo.swift]: Empty catch here, can we either handle it or use try? with a guard let

**Review by XanderSchoeman — APPROVED** [review]

**Pull request merged**